packages/swap/src/event-handlers/v140/depositBoosted.ts

```diff
@@ -0,0 +1,44 @@
+import { z } from 'zod';
+import { ONE_IN_PIP } from '@/shared/functions';
+import { actionSchema, internalAssetEnum, number, u128 } from '@/shared/parsers';
+import { EventHandlerArgs } from '..';
+
+export const depositBoostedSchema = z.object({
+  asset: internalAssetEnum,
+  amounts: z.array(z.tuple([number, u128])),
+  ingressFee: u128,
+  boostFee: u128,
+  action: actionSchema,
+  // channelId: u128,
+  // depositDetails: cfChainsBtcUtxoId,
+  // prewitnessedDepositId: u128,
+});
+
+// DepositBoosted event is emitted instead of NetworkDepositReceived event in v140 due to boost
+// We need to update the depositAmount and store the ingress fee just like we do in the NetworkDepositReceived event
+export const depositBoosted = async ({ prisma, event }: EventHandlerArgs) => {
+  const { asset, boostFee, action, ingressFee, amounts } = depositBoostedSchema.parse(event.args);
+
+  if (action.__kind === 'Swap') {
+    const depositAmount = amounts.reduce((acc, [_, amount]) => acc + amount, BigInt(0));
+    const effectiveBoostFeeBps = (boostFee * BigInt(ONE_IN_PIP)) / BigInt(depositAmount.toString());
+
+    await prisma.swap.update({
+      where: { nativeId: action.swapId },
+      data: {
+        effectiveBoostFeeBps: Number(effectiveBoostFeeBps),
+        depositAmount: depositAmount.toString(),
```

Comment:
do we need to override depositAmount here?
Can it happen that the initial deposit is different from the amount aggregate here?

Reply:
or we don't get DepositReceived for boost swaps at all? (I'm forgetting how it was)

Reply:
i think the `DepositBoosted` comes first (on boost) and then the `DepositReceived` once the safety margin is met (and boosters are repaid)

Reply:
> or we don't get DepositReceived for boost swaps at all? (I'm forgetting how it was)

Thats true, `DepositReceived` never comes in case of a boosted swap. Thats why i left the small comment above this handler to explain this. We basically need to do the same thing here as in the `DepositReceived` event, that is, update the `depositAmount` and insert the `ingressFee`.

Reply:
> i think the `DepositBoosted` comes first (on boost) and then the `DepositReceived` once the safety margin is met (and boosters are repaid)

`DepositReceived` was basically renamed to `DepositFinalised` and we already have a proxying handler for that.

Additionally, the `DepositFinalised` now has a different action for boosted swaps and normal swaps. For boosted swaps, the action will be `BoostersCredited`, which we ignore in the `depositReceived` handler. For normal swaps, the action is `Swap` which we handle in the `depositReceived` handler.
Due to that, we need to handle the same logic in the `depositReceived` handler regarding updating the `depositAmount` and `ingressFee`.